core/src/test/java/org/apache/iceberg/MetadataTableScanTestBase.java

```diff
@@ -44,9 +45,18 @@ protected static List<Object> parameters() {
   }
 
   protected Set<String> actualManifestListPaths(TableScan allManifestsTableScan) {
-    return StreamSupport.stream(allManifestsTableScan.planFiles().spliterator(), false)
-        .map(t -> (AllManifestsTable.ManifestListReadTask) t)
-        .map(t -> t.file().path().toString())
+    Function<FileScanTask, String> toManifestListPath = t -> t.file().path().toString();
+    return scanToPath(allManifestsTableScan, toManifestListPath);
+  }
+
+  protected Set<String> actualManifestPaths(TableScan entriesTableScan) {
```

Comment:
Sorry to add more , but how hard will it be to just combine these methods to something like scannedPaths